### DISCUSSION TITLE
local_json parameter, how specify soulbinds.json file?

### DISCUSSION THREAD

**Discussion by yurko13** [simulationcraft/simc#5896] [Q&A]
local_json parameter is used to retrieve character data from locally downloaded JSON formatted Blizzard Community API character object files.

The initial format of local_json was
`local_json=john.json,john`
`save=profile.simc`

After format was changed to
`local_json=mainfile,spec=specfile,equipment=equipmentfile`
`save=profile.simc`

Currently this second example works, but it does not import covenant/soulbinds info from the armory json files.
Is there a way to specify downloaded from armory soulbinds.json character file to have soulbind details retrieved and placed into profile.simc file for simulation?

**Comment by navv1234** [comment]
`soulbinds=soulbindsfile`

**Reply by yurko13** [reply]
Yesterday I tried to guess and tested this before posting this question.
Right now before writing this answer, I did test it again.
Unfortunately it did NOT add soulbinds info into saved file. 

It looks like I keep doing something wrong:

****
> ls -la char*

`-rw-r--r-- 1 user user 218093 Apr 17 15:12 char_equip.json`
`-rw-r--r-- 1 user user     88 Apr 17 15:14 char_import1.tmp`
`-rw-r--r-- 1 user user    118 Apr 17 15:11 char_import2.tmp`
`-rw-r--r-- 1 user user   5529 Apr 17 15:11 char_main.json`
`-rw-r--r-- 1 user user 218093 Apr 17 15:12 char_soulbinds.json`
`-rw-r--r-- 1 user user  70670 Apr 17 15:12 char_spec.json`

> cat char_import1.tmp

`local_json=char_main.json,spec=char_spec.json,equipment=char_equip.json`
`save=char1.simc`

> simc char_import1.tmp

`SimulationCraft 905-01 for World of Warcraft 9.0.5.38134 Live (hotfix 2021-03-31/38134, git build shadowlands 94514eb21a)`

`[2021-02-24]: PTR-Hotfix "Chain Lightning Spell Power Modifier was accidentally missed" for spell "Chain Lightning" effect #1 does not match verification value. Field: sp_coefficient, DBC: 0.635, Verify: 0.47`

`Generating profiles...`

> cat char_import2.tmp

`local_json=char_main.json,spec=char_spec.json,equipment=char_equip.json,soulbinds=char_soulbinds.json`
`save=char2.simc`

> simc char_import2.tmp

`SimulationCraft 905-01 for World of Warcraft 9.0.5.38134 Live (hotfix 2021-03-31/38134, git build shadowlands 94514eb21a)`

`[2021-02-24]: PTR-Hotfix "Chain Lightning Spell Power Modifier was accidentally missed" for spell "Chain Lightning" effect #1 does not match verification value. Field: sp_coefficient, DBC: 0.635, Verify: 0.47`

`Generating profiles...`

> ls -la char1.simc char2.simc

`-rw-r--r-- 1 user user 6627 Apr 17 15:16 char1.simc`
`-rw-r--r-- 1 user user 6657 Apr 17 15:16 char2.simc`

> diff char1.simc char2.simc

`3c3`
`< origin="char_main.json,spec=char_spec.json,equipment=char_equip.json"`
`---`
`> origin="char_main.json,spec=char_spec.json,equipment=char_equip.json,soulbinds=char_soulbinds.json"``>`

****

**Comment by seriallos** [comment]
I helped debug for a bit and it looks like if the soulbinds JSON is retrieved without specifying a locale, SimC can't determine the covenant. Once a local is specified, it all works. A quick look at the JSON makes it look like some `name` fields switch between a string and an array based on whether or not a locale is used in the query parameters and the array form may be what's tripping SimC up.

**Reply by yurko13** [reply]
The correct URL to retrieve soulbinds json data is
`https://us.api.blizzard.com/profile/wow/character/${SERVER}/${CHAR}/soulbinds?access_token=${TOKEN}&namespace=profile-us&locale=en_US`

I was doing it without `locale=en_US` parameter, as a result simc wasn't able to retrieve correct covenant/soulbinds data.

Thanks a lot for help to **seriallos** !
